core/trino-spi/src/main/java/io/trino/spi/expression/ConnectorExpressionVisitor.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi.expression;
+
+import javax.annotation.Nullable;
+
+public abstract class ConnectorExpressionVisitor<R, C>
```

Comment:
The choice not to have a visitor for ConnectorExpression was intentional. Visitors impose a rigid structure on the class hierarchy, and part of the goal of ConnectorExpression is to eventually allow it to be extended by connectors.

Reply:
I remember you were explaining this in one of my PRs as well. Maybe something we could document somewhere?

Reply:
Removed